electricitymap/contrib/capacity_parsers/REE.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime
+from logging import getLogger
+
+from requests import Response, Session
+
+from electricitymap.contrib.config import ZoneKey
+
+"""Disclaimer: Capacity for the Spanish isles is only avilable per archipelago."""
+logger = getLogger(__name__)
+MODE_MAPPING = {
+    "Hidráulica": "hydro",
+    "Turbinación bombeo": "hydro storage",
+    "Nuclear": "nuclear",
+    "Carbón": "coal",
+    "Fuel + Gas": "gas",
+    "Ciclo combinado": "gas",
+    "Eólica": "wind",
+    "Solar fotovoltaica": "solar",
+    "Solar térmica": "solar",
+    "Otras renovables": "unknown",
+    "Cogeneración": "gas",
+    "Residuos no renovables": "unknown",
+    "Residuos renovables": "biomass",
+    "Motores diésel": "oil",
+    "Turbina de gas": "gas",
+    "Turbina de vapor": "gas",
+}
+
+GEO_LIMIT_TO_GEO_IDS = {
+    "peninsular": 8741,
+    "canarias": 19,
+    "baleares": 18,
+    "ceuta": 12,
+    "melilla": 8746,
+}
+
+ZONE_KEY_TO_GEO_LIMIT = {
+    "ES": "peninsular",
+    "ES-IB-FO": "baleares",
+    "ES-IB-IZ": "baleares",
+    "ES-IB-MA": "baleares",
+    "ES-IB-ME": "baleares",
+    "ES-CN-FVLZ": "canarias",
+    "ES-CN-GC": "canarias",
+    "ES-CN-HI": "canarias",
+    "ES-CN-IG": "canarias",
+    "ES-CN-LP": "canarias",
+    "ES-CN-TE": "canarias",
+    "ES-CE": "ceuta",
+    "ES-ML": "melilla",
+}
+
+
+def fetch_production_capacity(zone_key: ZoneKey, target_datetime: datetime):
+    geo_limit = ZONE_KEY_TO_GEO_LIMIT[zone_key]
+    geo_ids = GEO_LIMIT_TO_GEO_IDS[geo_limit]
+    url = f"https://apidatos.ree.es/es/datos/generacion/potencia-instalada?start_date={target_datetime.strftime('%Y-01-01T00:00')}&end_date={target_datetime.strftime('%Y-12-31T23:59')}&time_trunc=year&geo_trunc=electric_system&geo_limit={geo_limit}&geo_ids={geo_ids}&tecno_select=all"
```

Comment:
would it make sense to split this url as a base_url and all the different headers? so it is a bit more readable?

Reply:
I would split this into the url and params and then use the params object in the request get function, much cleaner to look at since it's just dict.